packages/core/test/misc-tools.js

```diff
@@ -98,6 +98,89 @@ describe('Tools', () => {
+          '/storage/photos/73a6433488cca7c5bd6ed9836fd6d9c8.png","64x64":"https://cdn.zapier.com/storage/photos' +
+          '/73a6433488cca7c5bd6ed9836fd6d9c8.png","128x128":"https://cdn.zapier.com/storage/photos/73a6433488cc' +
+          'a7c5bd6ed9836fd6d9c8.png"},"name":"A Zapier App"}},"action":{"service":{"logos":{"16x16":"https://cd' +
+          'n.zapier.com/storage/photos/73a6433488cca7c5bd6ed9836fd6d9c8.png","32x32":"https://cdn.zapier.com/st' +
+          'orage/photos/73a6433488cca7c5bd6ed9836fd6d9c8.png","64x64":"https://cdn.zapier.com/storage/photos/73' +
+          'a6433488cca7c5bd6ed9836fd6d9c8.png","128x128":"https://cdn.zapier.com/storage/photos/73a6433488cca7c' +
+          '5bd6ed9836fd6d9c8.png"},"name":"A Zapier App"}}}}}',
+        length: 250,
+        expected:
+          '{"authData":{"api_key":"b81ed2f2-52c7-4c0b-80c7-f5142133172c"},"inputData":{},"inputDataRaw":{},"met' +
+          'a":{"isLoadingSample":false,"isFillingDynamicDropdown":false,"isTestingAuth":false,"isPopulatingDedu' +
+          'pe":false},"NOTE":"This data has been truncated."}',
+      },
+      {
+        value:
+          '[{"id":1,"first_name":"Alvis","last_name":"Baish","ip_address":"152.55.253.129"},{"id":2,"first_name' +
+          '":"Roxine","last_name":"Meah","ip_address":"194.125.26.25"},{"id":3,"first_name":"Olva","last_name":' +
+          '"Copins","ip_address":"61.163.179.10"},{"id":4,"first_name":"Thomas","last_name":"Jury","ip_address"' +
+          ':"63.200.70.181"},{"id":5,"first_name":"Leoline","last_name":"Barthorpe","ip_address":"82.41.102.177' +
+          '"},{"id":6,"first_name":"Benjamen","last_name":"Bynert","ip_address":"40.242.37.77"},{"id":7,"first_' +
+          'name":"Hillyer","last_name":"Brighouse","ip_address":"81.130.27.24"},{"id":8,"first_name":"Inger","l' +
+          'ast_name":"Bavin","ip_address":"159.53.220.53"},{"id":9,"first_name":"Brier","last_name":"Drover","i' +
+          'p_address":"128.160.157.77"},{"id":10,"first_name":"Garrett","last_name":"Broddle","ip_address":"229' +
+          '.80.187.137"},{"id":11,"first_name":"Marwin","last_name":"Espie","ip_address":"222.102.146.189"},{"i' +
+          'd":12,"first_name":"Lucian","last_name":"Catley","ip_address":"251.72.199.116"},{"id":13,"first_name' +
+          '":"Diarmid","last_name":"Scothorne","ip_address":"14.244.169.107"},{"id":14,"first_name":"Ricca","la' +
+          'st_name":"Ollerearnshaw","ip_address":"132.88.63.202"},{"id":15,"first_name":"Jacquenette","last_nam' +
+          'e":"Cowcha","ip_address":"163.126.95.174"}]',
+        length: 250,
+        expected:
+          '[{"id":15,"first_name":"Jacquenette","last_name":"Cowcha","ip_address":"163.126.95.174"},{"id":14,"f' +
+          'irst_name":"Ricca","last_name":"Ollerearnshaw","ip_address":"132.88.63.202"},{"id":13,"first_name":"' +
+          'Diarmid"},"NOTE: This data has been truncated."]',
```

Comment:
These tests are what I felt the least confident in - I grabbed some random data from a real log, spruced it up a bit and used that for my first test; the second test is from a random fake data generator. Both don't feel very... followable? But it's tricky because there's a pretty high "minimum" size to work with for `truncateData`; even if you use the minimum length of ~40, any truncated result will be `["NOTE: This data has been truncated."]` or `{"NOTE":"This data has been truncated."}`. You've got to use a minimum of like 100 to _see_ results. And large JSON objects take a ton of space due to prettier, so I tried stringifying them to save space. 😅

Reply:
Yeah, that's tricky. I think the json strings are a little harder to follow because of the lack of syntax highlighting, but I understand the rationale. Another option could be storing a big json in a file and reading it into the tests. If you had specific design considerations in mind, you could have something like

```json
{
    "a": {"smaller": true},
    "b": {"nested": {"ok": true}},
    "c": [{"list": true},1,2, null]
}
```

You could run general tests against the large root object and specific tests against the sub objects (so array-specific things could only worry about the `c` substructure. You could also use `[bigJson]` for array based things. That would give us one big json file that doesn't blow up the test size, but also can be more easily reasoned about.